### PR TITLE
deprecate and update voice regions

### PR DESCRIPTION
**Pull request by onerandomusername**
## Summary

- update list of valid voice regions
- deprecate guild.region

While this does have the breaking change tag, that is because the VoiceRegion enum was updated with the current voice regions. As some of these haven't existed in years, it *should* be fine.

## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint` or `pre-commit run --all-files`
- [ ] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [x] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Review by shiftinv — CHANGES_REQUESTED** [review]
The guild voice region is present in a few more places, ideally those should be deprecated as well:
- `Client.create_guild`
- `Template.create_guild`
- audit log change key (docs)

Could you also update the docs of `VoiceRegion` to match the new region list?

File: disnake/guild.py [hunk 1]
```diff
@@ -1014,6 +1011,21 @@ def member_count(self) -> int:
         except AttributeError:
             return len(self._members)
 
+    @property
+    def region(self) -> VoiceRegion:
+        """Optional[:class:`VoiceRegion`]: The region the guild belongs on. There is a chance that the region
+        will be a :class:`str` if the value is not recognised by the enumerator.
```
[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/guild.py:1017]: Its still possible to be a string if it not recognized, afaik

[the remark this answers is not in the document]
Comment (shiftinv) [reply disnake/guild.py:1017]: `try_enum` dynamically creates an enum value if it doesn't exist, so this should never be a `str`

[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/guild.py]: https://github.com/DisnakeDev/disnake/pull/357/commits/1c910a52a49f51f6632aad4e2dfa6fb024e4e777

File: disnake/guild.py [hunk 2]
```diff
@@ -1705,6 +1721,9 @@ async def edit(
             fields["owner_id"] = owner.id
 
         if region is not MISSING:
+            utils.warn_deprecated(
+                "Guild.region is deprecated and will be removed in version 2.6.", stacklevel=2
+            )
             fields["region"] = str(region)
```
[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/guild.py:1727]: I think its fine to leave in HTTPClient as while its deprecated, the api still will accept it? Not sure.

[the remark this answers is not in the document]
Comment (onerandomusername) [reply disnake/guild.py]: Removed https://github.com/DisnakeDev/disnake/pull/357/commits/555c63be4e474e3a03865c14105320f304d01a4e

**Review by shiftinv — APPROVED** [review]

**Pull request merged**